File: cpp/3024_triangle_type/triangle_type.cpp
#include <algorithm>
#include <cassert>
#include <string>
#include <tuple>
#include <vector>

using namespace std;

class Solution {
public:
    string triangleType(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        if (nums[0] + nums[1] <= nums[2]) {
            return "none";
        } else if (nums[0] == nums[2]) {
            return "equilateral";
        } else if (nums[0] == nums[1] || nums[1] == nums[2]) {
            return "isosceles";
        } else {
            return "scalene";
        }
    }
};

int main() {
  vector<tuple<vector<int>, string>> tests {
    {{3,3,3}, "equilateral"},
    {{3,4,5}, "scalene"},
  };

  for (auto &[nums, ans] : tests) {
    assert(Solution().triangleType(nums) == ans);
  }
}

File: golang/3024_triangle_type/triangle_type.go
// Package main ...
package main

import (
	"sort"
	"testing"

	"github.com/stretchr/testify/assert"
)

func triangleType(nums []int) string {
	sort.Ints(nums)

	switch {
	case nums[0]+nums[1] <= nums[2]:
		return "none"
	case nums[0] == nums[2]:
		return "equilateral"
	case nums[0] == nums[1] || nums[1] == nums[2]:
		return "isosceles"
	default:
		return "scalene"
	}
}

func main() {
	tests := []struct {
		nums []int
		ans  string
	}{
		{[]int{3, 3, 3}, "equilateral"},
		{[]int{3, 4, 5}, "scalene"},
	}

	for index, test := range tests {
		assert.Equal(&testing.T{}, test.ans, triangleType(test.nums), index)
	}
}
